### PR TITLE
Feat: Add `external_aliases` Which Will Not Have CNAMEs Created for Them

### PR DESCRIPTION
**Pull request by mburns**
## what
* Allow for aliases in CloudFront which do not get the corresponding CNAME record created in Route53.
* Misc: add BridgeCrew suppressions.

## why
* Not all CloudFront domains are managed in Route53
* This is similar to `dns_alias_enabled`, but allows for a mixture of DNS providers for a single CloudFront Distribution
* Some new false positives were raised by BridgeCrew, hence requiring some suppressions.

### CODE REVIEW

**Comment by korenyoni** [comment]
@mburns are you able to please add BridgeCrew exceptions for the failing checks outlined in this PR?

File: main.tf [hunk 1]
```diff
@@ -270,7 +270,7 @@ resource "aws_s3_bucket" "origin" {
   }
 
   dynamic "cors_rule" {
-    for_each = distinct(compact(concat(var.cors_allowed_origins, var.aliases)))
+    for_each = distinct(compact(concat(var.cors_allowed_origins, var.aliases, var.external_aliases)))
```
[the remark this answers is not in the document]
Comment (korenyoni) [reply main.tf:273]: Please add BridgeCrew exceptions for these errors

File: main.tf [hunk 2]
```diff
@@ -338,7 +338,7 @@ resource "aws_cloudfront_distribution" "default" {
     }
   }
 
-  aliases = var.acm_certificate_arn != "" ? var.aliases : []
+  aliases = var.acm_certificate_arn != "" ? concat(var.aliases, var.external_aliases) : []
```
[the remark this answers is not in the document]
Comment (korenyoni) [reply main.tf:341]: Same here — please add a BridgeCrew exception

**Review by korenyoni — CHANGES_REQUESTED** [review]
LGTM. But please see comments regarding BridgeCrew.

**Comment by korenyoni** [comment]
Hey @mburns — I see that you reverted the BridgeCrew exceptions, was there a particular reason why?

**Comment by mburns** [comment]
HI @korenyoni, just troubleshooting, sorry. It looked like the errors were fixed with merging the latest master.

**Comment by korenyoni** [comment]
@mburns I'm honestly not sure why the BridgeCrew check is still failing — I will have to take a closer look tomorrow.

CleanShot 2021-12-14 at 00 57 21@2x

**Comment by korenyoni** [comment]
CleanShot 2021-12-14 at 12 35 31@2x

I'm still not sure why BC is failing in the GitHub checks API.

Originally, it was because `examples/complete`'s module instantiation was showing up in BridgeCrew without the suppressions: 
CleanShot 2021-12-14 at 12 09 38@2x

I'm not sure why it's still happening though. As you can see, it's passing in BC itself.

CleanShot 2021-12-14 at 12 35 47@2x

@mburns I've also taken over your PR and just expanded the suppression comments and moved the suppressions to the top of the resource definition blocks.

Anyways, I'm going to merge this in and hope the BC errors do not keep happening in subsequent PRs.

**Review by korenyoni — APPROVED** [review]

**Pull request merged**